src/open-uri.c

```diff
@@ -586,7 +586,7 @@ handle_open_in_thread_func (GTask *task,
   gboolean ask_for_content_type;
   GVariantBuilder opts_builder;
   gboolean skip_app_chooser = FALSE;
-  int fd;
+  g_auto(XdpFd) fd = -1;
```

Comment:
I was looking through the post-1.8.0 changes in the repo, and this change looked a bit suspect.

In particular, this function still has a `close(fd)` call buried in its body, so the destructor will sometimes call close() on a stale (or reallocated) file descriptor.  Might it be better to manage the lifetime by storing the fd with g_object_set_data_full() with an appropriate destroy_notify function (i.e. something that calls close() on values >= 0).